homeassistant/components/green_planet_energy/__init__.py

```diff
@@ -0,0 +1,109 @@
+"""Green Planet Energy integration for Home Assistant."""
+
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+)
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import GreenPlanetEnergyUpdateCoordinator
+from .sensor import get_current_price
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+# Config entry only (no YAML config)
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+# Service schema
+SERVICE_GET_PRICE_SCHEMA = vol.Schema(
+    {
+        vol.Optional("hour"): vol.All(vol.Coerce(int), vol.Range(min=0, max=23)),
+    }
+)
```

Comment:
Oh, maybe I should have added this, but we should do one platform per PR to make it still easy to review. So let's split the service part off for now, and add that in a subsequent PR